### DISCUSSION TITLE
Running vtracer on macOS

### DISCUSSION THREAD

**Discussion by grahammiln** [visioncortex/vtracer#18] [Show and tell]
On macOS, the [pre-compiled binary](https://github.com/visioncortex/vtracer/releases) needs a few local changes before it can be launched.

In Terminal.app, run these two commands:

```
chmod +x vtracer-mac
codesign -s - vtracer-mac
```

The first command, `chmod`, sets the file as executable. The second, `codesign`, code signs the binary with an ad-hoc signature.

Finally, **Control-click** or **right-click** on the binary's icon in the Finder and select **Open**. This will present a warning to override macOS's Gatekeeper.

With these steps done, the binary can now be launched from the terminal.

To avoid these steps, released binaries need to be [code signed and notarised](https://support.apple.com/en-gb/guide/security/sec3ad8e6e53/web) by a member of the [Apple Developer Program](https://developer.apple.com/programs/).

Hope this helps!

**Comment by tyt2y3** [comment]
Thanks! Let me make a M1 Mac build too.
Though I have no interest in joining the Apple Dev Program
